engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnStateTransitionBehavior.java

```diff
@@ -155,28 +155,36 @@ private void verifyIncidentResolving(final BpmnElementContext context, final Str
     }
   }
 
-  public <T extends ExecutableFlowNode>
-      BpmnElementContext transitionToCompletedWithParentNotification(
-          final T element, final BpmnElementContext context) {
-    final boolean endOfExecutionPath = element.getOutgoing().isEmpty();
+  /** @return context with updated intent */
+  public <T extends ExecutableFlowNode> Either<Failure, BpmnElementContext> transitionToCompleted(
+      final T element, final BpmnElementContext context) {
+    final boolean endOfExecutionPath;
+    if (context.getBpmnElementType() == BpmnElementType.PROCESS) {
+      // a completing child process is not considered here.
+      // the corresponding call activity is informed of the
+      // child process completion explicitly by the process processor
+      endOfExecutionPath = false;
+    } else {
+      endOfExecutionPath = element.getOutgoing().isEmpty();
+    }
 
     if (endOfExecutionPath) {
-      beforeExecutionPathCompleted(element, context);
+      final Either<Failure, ?> failureOrSuccess = beforeExecutionPathCompleted(element, context);
+      if (failureOrSuccess.isLeft()) {
+        return failureOrSuccess.map(ok -> context);
+      }
```

Comment:
(optional) Reduce confusion by applying the map() function before. It looks odd to call map() if the `either` is left.

```suggestion
      final var failureOrSuccess =
          beforeExecutionPathCompleted(element, context).map(ok -> context);
      if (failureOrSuccess.isLeft()) {
        return failureOrSuccess;
      }
```